### ISSUE TITLE
Add support for using host network for porter actions

### ISSUE THREAD

**Issue by jasmdk** [getporter/porter#1347]
In our CI pipeline, we would like to spin up a k3d cluster and deploy a set of bundles, however we rely on different hostnames to be routed to different services, but in a CI scenario we can not control external DNS so the host names are constructed. I can add host mappings for each hostname either in the hosts file of the k3d nodes or even on the build agent where my CI is happening, but when I run the invocation image on the agent, then it do not get access to the hosts defined on the agent.
My current issue is that I am using the Hashicorp Vault CLI to configure vault, but as the vault is expected to be found at host vault.<some-base-domain> then it fails to resolve the host name.

I propose to control the --network=host flag of the docker execution by a "use-docker-host-network" parameter or similar which will ensure that the /etc/hosts on the docker host will be taken into account.

Alternatives considered:
Add support adding explicit host entries like --add-host for docker command.
Use the cnab-go kubernetes driver instead of the docker driver, but it is not available through porter.
Configuring extra dns servers on each docker agent where porter should be working.

**Comment by carolynvs** [comment]
@jasmdk As a workaround, would copying a hosts file into the bundle help at all? For example, the bundle can define a parameter that would let you pass in a hosts file with the necessary entries:

```yaml
parameters:
- name: hosts
   type: file
   path: /etc/hosts
```

**Comment by jasmdk** [comment]
I agree that it probably could do the trick, however we found another solution, using a public dns with a mapping to an internal ip. This also allows us to create SSL certificates with letsencrypt that are trusted without having to register the certificates explicitly on the docker daemons.

**Comment by carolynvs** [comment]
Nice! I've used services like that for local dev, never thought of using it for something like this too. 👍 

Keeping this open because eventually we want to support this without workarounds.

**Comment by cgiraldo** [comment]
Just to add here my use case:

I am just trying things in a microk8s, however porter.sh does not work with microk8s.kubeconfig (kubernetes api host is set to 127.0.0.1). 
I have updated my kubeconfig and microk8s certs with a hostname instead of 127.0.0.1, but the hostname must be resolvable in the docker container that Porter runs. updating /etc/hosts make my local kubectl work, but not inside porter container.

**Comment by vdice** [comment]
Thanks for adding your use case,  @cgiraldo .  For the /etc/hosts approach to work, I believe the same updated file needs to exist inside the Porter container to work.  Have you tried adding it in as a parameter as Carolyn mentions [above](https://github.com/getporter/porter/issues/1347#issuecomment-718817767)?

**Comment by cgiraldo** [comment]
I prefer not to include tricks in the bundle. 

I have set up `192-168-0-3.nip.io` in my kubeconfig api_url hostname and that did the trick for me.

The hostname in kubeconfig is important for me because I build ingress_url outputs based on this hostname.

FYI:
```
web_ui_info() {
    NAMESPACE=$(kubectl config view --minify=true --output jsonpath={$.contexts[0].context.namespace})
    RELEASE=$CNAB_INSTALLATION_NAME
    DOMAIN=$(kubectl config view --minify=true --output jsonpath={$.clusters[0].cluster.server} | sed 's@https://@@' | sed 's@http://@@' | sed 's/:.*//')
    echo "ingress_url: https://$RELEASE.$NAMESPACE.$DOMAIN"
}
```

**Comment by carolynvs** [comment]
Closing since it sounds like there is an acceptable workaround. One other way we've learned of recently to refer to the IP of the host (for docker for mac/windows) is to use `host.docker.internal` in the kubeconfig file.

**Comment by carolynvs** [comment]
Oops, actually let's keep this open and document the two known workarounds.
